Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs

```diff
@@ -472,7 +473,11 @@ public async Task ProcessExecutionResult(SubmissionExecutionResult submissionExe
 
             this.submissionsData.Update(submission);
 
-            await this.UpdateResults(submission);
+            await this.SaveParticipantScore(submission);
+
+            await this.submissionsForProcessingData.MarkProcessed(submission.Id);
+            await this.submissionsData.SaveChanges();
+            CacheTestRuns(submission);
```

Comment:
The submission should not be marked as processed before the TestRuns are cached, as this may fail, thus leaving not fully processed submission marked as processed.

Reply:
The submission will not be marked as Processed because, this logic is in a transaction scope. Therefore nothing will be saved in the database until the transaction scope is completed.